### PR TITLE
stop parsing errors from crashing the whole page

### PR DESCRIPTION
**Pull request by LeoMcA**
https://bugzilla.mozilla.org/show_bug.cgi?id=1762414

### CODE REVIEW

File: kitsune/sumo/parser.py [hunk 1]
```diff
@@ -244,16 +245,22 @@ def parse(
...
+                    text,
+                    show_toc=show_toc,
+                    tags=tags or ALLOWED_TAGS,
+                    attributes=attributes or ALLOWED_ATTRIBUTES,
+                    styles=styles or ALLOWED_STYLES,
+                    nofollow=nofollow,
+                    strip_comments=True,
+                    **kwargs,
+                )
+            except Exception as e:
```
Comment (akatsoulas) [inline kitsune/sumo/parser.py:259]: let's make this a `TypeError`

**Review by akatsoulas — APPROVED** [review]
r+wc

File: kitsune/sumo/parser.py (hunk 1 above)
Comment (LeoMcA) [reply kitsune/sumo/parser.py:259]: I wanted to catch all errors, not just the specific one in the bug above, to guard against any future parsing bugs we may encounter.

Comment (akatsoulas) [reply kitsune/sumo/parser.py:259]: let's catch the one we know about instead of silencing everything.

**Pull request merged**